### PR TITLE
First lambda arg is on the line with brace

### LINKED ISSUE #94: slight formatting issue: brace is left alone in args
## Input
```nix
{foo,
bar}:
              foo+bar
```

## Output
```nix
{
  foo
, bar
}:
foo + bar

```

## Desired output

This seems more canonical in nixpkgs:

```nix
{ foo
, bar
}:
foo + bar
```

### PR DESCRIPTION
**Pull request by matklad**
Closes #94 

We always format lambdas as

```
{ a
, b }:
```

To make case like

```
{
  foo = { a
        , b }:
}
```

pretty, we just force lambda onto the new line

### CODE REVIEW

**Comment by matklad** [comment]
@zimbatm I am not sure we actually have consensus on what's the right behavior here, so I've implemented something which seemed reasonable :)

File: src/rules.rs [hunk 1]
```diff
@@ -94,6 +95,20 @@ pub(crate) fn spacing() -> SpacingDsl {
         .inside(NODE_IF_ELSE).after(T![if]).single_space_or_optional_newline()
         .inside(NODE_IF_ELSE).around([T![then], T![else]]).single_space_or_optional_newline()
 
+        // special-case to force a linebreak before `=` in
+        //
+        // ```nix
+        // {
+        //   long_key = { x
+        //              , y
+        //              , z}: body
```
Comment (zimbatm) [inline src/rules.rs]:
```suggestion
        //              , z
        //              }: body
```
the closing curly should be on it's own line.

**Comment by zimbatm** [comment]
Yeah that seems reasonable.

**Pull request merged**